scripts/server.py

```diff
@@ -65,6 +64,48 @@ def search():
                            env=app.config.get('ENV', None))
 
 
+@app.route('/clusters/', methods=['GET'])
+def clusters():
+    # use a facet pivot query to get tags that occur together on at
+    # at least 50 documents
+    sqs = SolrQuerySet(get_solr()) \
+        .facet('tags_ss', pivot='tags_ss,tags_ss', **{'pivot.mincount': 50})
+    facets = sqs.get_facets()
+
+    # iterate over the facet pivots to generate a unique list of pairs & counts
+    tag_pairs = []
+    tag_pair_counts = {}
+    for tag_pivot in facets.facet_pivot['tags_ss,tags_ss']:
+        tag_a = tag_pivot['value']
+        for pivot in tag_pivot['pivot']:
+            tag_b = pivot['value']
+            # every pivot repeats the first tag; skip that one
+            if tag_b == tag_a:
+                continue
+            tag_pair = '/'.join([tag_a, tag_b])
+            # each pair will show up twice; only add it the first time
+            alt_tag_pair = '/'.join([tag_b, tag_a])
+            if alt_tag_pair in tag_pairs:
+                continue
+            tag_pairs.append(tag_pair)
+            tag_pair_counts[tag_pair] = pivot['count']
```

Comment:
there might be a more efficient way to do this, but certainly not necessary to figure out now. it's weird how much information solr repeats in the response!

Reply:
@thatbudakguy I think the redundancy is because you usually pivot on different fields, not the same one. (Or at least, that's my sense of the typical use case for the functionality)

Reply:
ah, that makes a lot of sense. I guess this is kind of a strange case